packages/core/src/lib/Select/TraditionalSelect/TraditionalSelect.tsx

```diff
@@ -30,48 +30,65 @@ const TraditionalSelect: React.FC<ISelectProps> = ({
     width = '200px',
     ...props
 }: ISelectProps) => {
-    const defaultValue =
-        (defaultOptionIndex !== undefined &&
-            options[defaultOptionIndex]?.label) ||
-        value ||
-        placeholder ||
-        'Please choose';
+    const [currentValue, setCurrentValue] = useState<
+        string | number | string[]
+    >();
+
+    useEffect(() => {
+        if (!value) return;
+        setCurrentValue(value);
+    }, [value]);
+
+    useEffect(() => {
+        if (Number(defaultOptionIndex) < 0) return;
+        if (Number(defaultOptionIndex) > options.length) return;
+
+        setCurrentValue(options[Number(defaultOptionIndex)]?.label || '');
+    }, [defaultOptionIndex]);
+
+    const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+        setCurrentValue(event.currentTarget.value);
+        onChangeTraditional && onChangeTraditional(event);
+    };
+
     return (
         <DivWrapperStyled
-            className="input_filled"
+            className="input_filled select"
             data-testid="test-select"
+            state={state}
             style={{ ...style, width }}
             {...props}
         >
             <SelectStyled
                 data-testid="test-select-select"
-                defaultValue={defaultValue}
+                disabled={disabled}
                 id={id}
                 ref={refTraditional}
                 onChange={(event: React.ChangeEvent<HTMLSelectElement>) =>
-                    onChangeTraditional && onChangeTraditional(event)
+                    handleChange(event)
                 }
                 onBlur={(event: React.FocusEvent<HTMLSelectElement>) =>
                     onBlurTraditional && onBlurTraditional(event)
                 }
                 required={validation?.required}
-                value={value}
+                value={currentValue || placeholder || 'Please choose'}
             >
                 <option disabled>{placeholder || 'Please choose'}</option>
 
                 {options.map((option, i) => (
                     <option
                         data-testid={`test-select-option-${i}`}
+                        disabled={option.disabled || false}
```

Comment:
there is no style indicating that the component is disabled. 

![image](https://user-images.githubusercontent.com/35369843/221425572-13589651-e2d8-4d01-87a0-183b09f8c17e.png)
This is the disabled component and looks like it is not disabled. but on clicking it nothing happens. I think we do have designs for it